vaas-app/src/vaas/manager/signals.py

```diff
@@ -160,9 +160,11 @@ def vcl_update(sender, **kwargs):
     # Route
     elif sender is Route:
         for route in Route.objects.all():
-            if route.cluster == instance.cluster:
-                logger.debug("vcl_update(): %s" % str(route.cluster))
-                clusters_to_refresh.append(route.cluster)
+            for cluster in route.clusters.all():
```

Comment:
instead of making nested db-queries - please use [prefetch_related](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#prefetch-related) to optimize db usage